tests/container-queries.test.tsx

```diff
@@ -0,0 +1,367 @@
+      width: 160,
+    });
+  });
+
+  it('supports basic container queries', () => {
+    const StyledComponent = styled.Text`
+      font-size: 2rem;
+      @container (max-width: 800px) {
+        font-size: 1.5rem;
+      }
+    `;
+
+    render(
+      <StyledFunctionContainer width={1000}>
+        <StyledComponent />
+      </StyledFunctionContainer>
+    );
+    sleep(1000);
+    expect(Text).toHaveBeenLastCalledWith(
+      expect.objectContaining({ style: { fontSize: 2 * theme.rem } }),
+      {}
+    );
+
+    render(
+      <StyledFunctionContainer width={400}>
+        <StyledComponent />
+      </StyledFunctionContainer>
+    );
+    sleep(1000);
+    expect(Text).toHaveBeenLastCalledWith(
+      expect.objectContaining({ style: { fontSize: 1.5 * theme.rem } }),
+      {}
+    );
+  });
+
+  it('does not render the query without a defined container', () => {
+    const StyledNotContainer = styled.View<{ width: number }>`
+      width: ${(p) => p.width}px;
+    `;
+
+    const StyledComponent = styled.Text`
+      font-size: 2rem;
+      @container (max-width: 800px) {
+        font-size: 1.5rem;
+      }
+    `;
+
+    render(
+      <StyledNotContainer width={1000}>
+        <StyledComponent />
+      </StyledNotContainer>
+    );
+    sleep(1000);
+    expect(Text).toHaveBeenLastCalledWith(
+      expect.objectContaining({ style: { fontSize: 2 * theme.rem } }),
+      {}
+    );
+
+    render(
+      <StyledNotContainer width={500}>
+        <StyledComponent />
+      </StyledNotContainer>
+    );
+    sleep(1000);
+    expect(Text).toHaveBeenLastCalledWith(
+      expect.objectContaining({ style: { fontSize: 2 * theme.rem } }),
+      {}
+    );
+  });
+
+  it('does render with function parameters in the contained element', () => {
+    const StyledComponent = styled.Text<{ width: number }>`
+      width: ${(p) => p.width}px;
+      font-size: 2rem;
+      @container (min-height: 800px) {
+        font-size: 1.5rem;
+      }
+    `;
+
+    render(
+      <StyledFunctionContainer width={1000} height={500}>
+        <StyledComponent width={100} />
+      </StyledFunctionContainer>
+    );
+    sleep(1000);
+    expect(Text).toHaveBeenLastCalledWith(
+      expect.objectContaining({ style: { fontSize: 2 * theme.rem, width: 100 } }),
+      {}
+    );
+
+    render(
+      <StyledFunctionContainer width={1000} height={1000}>
+        <StyledComponent width={100} />
+      </StyledFunctionContainer>
+    );
+    sleep(1000);
+    expect(Text).toHaveBeenLastCalledWith(
+      expect.objectContaining({ style: { fontSize: 1.5 * theme.rem, width: 100 } }),
+      {}
+    );
+  });
+
+  it('supports container and media queries', () => {
+    const StyledMediaComponent = styled.Text`
+      font-size: 2rem;
+      @media (max-width: 800px) {
+        font-size: 1.5rem;
+      }
+    `;
+
+    //both cases render font size 1.5rem even with the container size changing
```

Comment:
I would add a case here that tests that the media query still fires correctly with the same way that is mocked in the media query tests